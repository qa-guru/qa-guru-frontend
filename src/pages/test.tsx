import { FC } from "react";
import { useParams } from "react-router-dom";

import TestContainer from "features/test/containers/test-container";

const TestPage: FC = () => {
  const { testId, trainingId, lectureId } = useParams<{
    testId: string;
    trainingId: string;
    lectureId: string;
  }>();

  // Проверяем, что все параметры присутствуют
  if (!testId || !trainingId || !lectureId) {
    return <div>Ошибка: отсутствуют необходимые параметры</div>;
  }

  return (
    <TestContainer
      testId={testId}
      trainingId={trainingId}
      lectureId={lectureId}
    />
  );
};

export default TestPage;
