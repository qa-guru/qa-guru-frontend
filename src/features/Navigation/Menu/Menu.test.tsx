import { Link, MemoryRouter } from "react-router-dom";
import { render } from "../../../test/utilsTest";
import AppMenu from "./Menu";
import * as React from "react";

const mockHandleClickNavMenu = () => {};

const mockPages = [
  {
    title: <Link to="">{""}</Link>,
    pageURL: "",
  },
];

describe("Menu", () => {
  it("the component is render", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <AppMenu
          handleClickNavMenu={mockHandleClickNavMenu}
          pages={mockPages}
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
