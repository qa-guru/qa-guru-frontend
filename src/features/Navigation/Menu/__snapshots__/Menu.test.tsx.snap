// Vitest Snapshot v1

exports[`Menu > the component is render 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-j0f50d"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
      >
        <a
          href="/"
        />
      </h6>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
